File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/view/MainActivity.java
package com.mobiledevpro.apptemplate.ui.mainscreen.view;

import android.view.View;

import com.mobiledevpro.apptemplate.R;
import com.mobiledevpro.apptemplate.helper.FragmentsHelper;
import com.mobiledevpro.commons.activity.BaseActivity;

import androidx.appcompat.widget.Toolbar;

public class MainActivity extends BaseActivity {

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        if (toolbar != null) {
            setSupportActionBar(toolbar);
        }
    }

    @Override
    protected void initPresenters() {
        //no need
    }

    @Override
    protected void populateView(View layoutView) {
        //work with view
        FragmentsHelper.showMainFragment(mFragmentManager, R.id.fragment_container);
    }
}

File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/view/MainActivity.kt
package com.mobiledevpro.apptemplate.ui.mainscreen.view

import android.view.View
import androidx.appcompat.widget.Toolbar
import com.mobiledevpro.apptemplate.R
import com.mobiledevpro.commons.activity.BaseActivity

class MainActivity : BaseActivity() {

    override fun getLayoutResId() = R.layout.activity_main

    override fun initToolbar() {
        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
    }

    override fun initPresenters() {
        //no need
    }

    override fun populateView(layoutView: View) {
        //work with view

    }
}
